### PR TITLE
Add Library Reference placeholder page

### PR DESCRIPTION
**Pull request by arsci**
* Adds Library Reference placeholder page (eventually this will be replaced by the filterable module search)
* Swap Modules and Services in sidebar per Eben's feedback

### CODE REVIEW

File: _docs-sources/iac/reference/index.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
...
+The Library Reference serves as the definitive source for all of the Modules and Services within the Gruntwork Infrastructure as Code Library. This comprehensive reference provides a dedicated page for each module and service providing descriptions, detailed information on input and output variables, and sample code to help you get started.
```
Comment (oredavids) [inline _docs-sources/iac/reference/index.md:3]:
```suggestion
The Library Reference serves as the definitive source for all actively maintained Modules and Services within the Gruntwork Infrastructure as Code Library. This comprehensive reference provides a dedicated page for each module and service providing descriptions, detailed information on input and output variables, and sample code to help you get started.
```

Suggestion based on the fact that some customers have access to deprecated modules that do not show up within References.

Comment (ellisonc) [inline _docs-sources/iac/reference/index.md:3]: "definitive source" sounds strange to me. Maybe definitive reference? index? documentation? IDK, words are hard

**Review by oredavids — APPROVED** [review]
Looks good. Left one small suggestion.

File: _docs-sources/iac/reference/index.md (hunk 1 above)
Comment (arsci) [reply _docs-sources/iac/reference/index.md:3]: I think `index` does sound better. Reading it back to myself, `source` would not be the right word anyway as GitHub is really the definitive source here, and I think it sounds a bit odd stating reference again.

`The Library Reference serves as the definitive index for all of the Modules and Services...` seems to work

**Review by MoonMoon1919 — APPROVED** [review]
lgtm

**Pull request merged**